src/utilities/types/DeepOmit.ts

```diff
@@ -0,0 +1,17 @@
+import { Primitive } from './Primitive';
+
+export type DeepOmitArray<T extends any[], K> = {
+  [P in keyof T]: DeepOmit<T[P], K>;
+};
+
+export type DeepOmit<T, K> = T extends Primitive
+  ? T
+  : {
+      [P in Exclude<keyof T, K>]: T[P] extends infer TP
+        ? TP extends Primitive
+          ? TP
+          : TP extends any[]
+          ? DeepOmitArray<TP, K>
+          : DeepOmit<TP, K>
+        : never;
+    };
```

Comment:
```suggestion
export type DeepOmit<T, K> = T extends Primitive
  ? T
  : {
      [P in Exclude<keyof T, K>]: T[P] extends infer TP
        ? TP extends Primitive
          ? TP
          : TP extends any[]
            ? DeepOmitArray<TP, K>
            : DeepOmit<TP, K>
        : never;
    };
```
A little more indentation helps readability here, and seems consistent with your other `?:` indentation.

Reply:
While I agree this change would be better, I'm beholden to prettier here and just let it do its auto formatting. I'll consider a `// prettier-ignore`.

Reply:
Unfortunately it looks like `// prettier-ignore` isn't much bettter. It keeps wanting to move the `?` before the comment.

```ts
export type DeepOmit<T, K> = T extends Primitive
  ? T
  : {
      [P in Exclude<keyof T, K>]: T[P] extends infer TP
        ? TP extends Primitive
          ? TP
          : TP extends any[]
            // prettier-ignore
            ? DeepOmitArray<TP, K>
            : DeepOmit<TP, K>
        : never;
    };
```

Output:

```ts
export type DeepOmit<T, K> = T extends Primitive
  ? T
  : {
      [P in Exclude<keyof T, K>]: T[P] extends infer TP
        ? TP extends Primitive
          ? TP
          : TP extends any[]
          ? // prettier-ignore
            DeepOmitArray<TP, K>
          : DeepOmit<TP, K>
        : never;
    };

```

I could add it above the `extend type DeepOmit<T, K>`, but that negates all formatting for the type (I'd like to avoid removing all formatting for that block if possible).

Thoughts?

Reply:
Not worth fighting Prettier over, I guess. Comments attachment is tricky. You might be able to insert some parentheses to influence the wrapping? Or maybe `/* prettier-ignore */` comments work differently?

Reply:
None of it works 😩. Prettier removes the parentheses and `/* prettier-ignore */` works the same as `// prettier-ignore`. Such a bummer. For now I'm going to leave this as-is.